.github/workflows/test.yml

```diff
@@ -74,5 +74,5 @@ jobs:
       - name: test upgrade
         run: |
           chmod +x -R ./contrib/updates
-          ./contrib/updates/upgrade-test.sh
+          ./contrib/updates/upgrade-test.sh 7
```

Comment:
Is 7 a magical number of the number of vals your spinning up? Maybe source this from a ENV (TESTNET_NVAL) variable and then default to 7 if the ENV var is empty?